tests/PHPStan/Rules/Variables/IssetRuleTest.php

```diff
@@ -450,4 +450,21 @@ public function testObjectShapes(): void
 		$this->analyse([__DIR__ . '/data/isset-object-shapes.php'], []);
 	}
 
+	public function testBug3985(): void
+	{
+		$this->treatPhpDocTypesAsCertain = true;
+		$this->strictUnnecessaryNullsafePropertyFetch = true;
+
+		$this->analyse([__DIR__ . '/../../Analyser/data/bug-3985.php'], [
```

Comment:
I decided to add new assertions for this old bug, while in the process of understanding the internals better
(IMO the new assertions better describe what the end user expected)